javapoet/src/main/java/com/palantir/javapoet/MethodSpec.java

```diff
@@ -133,7 +133,8 @@ private boolean lastParameterIsArray(List<ParameterSpec> parameters) {
                 && TypeName.asArray(parameters.get(parameters.size() - 1).type()) != null;
     }
 
-    void emit(CodeWriter codeWriter, String enclosingName, Set<Modifier> implicitModifiers) throws IOException {
+    void emit(CodeWriter codeWriter, String enclosingName, Set<Modifier> implicitModifiers, boolean compactConstructor)
```

Comment:
Instead of passing this `compactConstructor` from the `TypeSpec`, let's just make this part of the `MethodSpec`.

I'm imagining the `builder` methods would look something like:

```java
public static Builder methodBuilder(String name) {
    return new Builder(name, false);
}

public static Builder constructorBuilder() {
    return new Builder(CONSTRUCTOR, false);
}

public static Builder compactConstructorBuilder() {
    return new Builder(CONSTRUCTOR, true);
}
```